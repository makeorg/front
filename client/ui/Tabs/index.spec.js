import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { Tabs } from './index';

const defaultProps = [
  {
    tab: 'fooTab',
    panel: 'fooPanel',
  },
  {
    tab: 'barTab',
    panel: 'barPanel',
  },
];

const disabledTabProps = [
  {
    tab: 'fooTab',
    panel: 'fooPanel',
    isDisabled: true,
  },
  {
    tab: 'barTab',
    panel: 'barPanel',
    isDisabled: true,
  },
];

describe('Tabs', () => {
  it('toMatchSnapshot', () => {
    const component = renderer
      .create(<Tabs tabsContent={defaultProps} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must return the diff between snapshot when Tab is enabled vs disabled', () => {
    const enabledTabs = renderer
      .create(<Tabs tabsContent={defaultProps} />)
      .toJSON();
    const disabledTabs = renderer
      .create(<Tabs tabsContent={disabledTabProps} />)
      .toJSON();
    expect(snapshotDiff(enabledTabs, disabledTabs)).toMatchSnapshot();
  });
});
