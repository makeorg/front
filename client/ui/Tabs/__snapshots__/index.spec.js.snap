// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs must return the diff between snapshot when Tab is enabled vs disabled 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -7,42 +7,42 @@
    <ul
      className="sc-1c1pe0e-1 gqwjDh"
      role="tablist"
    >
      <li
-       aria-controls="react-tabs-5"
-       aria-disabled="false"
+       aria-controls="react-tabs-9"
+       aria-disabled="true"
        aria-selected="true"
-       className="sc-1c1pe0e-2 kjwiMK selected"
-       id="react-tabs-4"
+       className="sc-1c1pe0e-2 kjwiMK selected disabled"
+       id="react-tabs-8"
        role="tab"
        tabIndex="0"
      >
        fooTab
      </li>
      <li
-       aria-controls="react-tabs-7"
-       aria-disabled="false"
+       aria-controls="react-tabs-11"
+       aria-disabled="true"
        aria-selected="false"
-       className="sc-1c1pe0e-2 kjwiMK"
-       id="react-tabs-6"
+       className="sc-1c1pe0e-2 kjwiMK disabled"
+       id="react-tabs-10"
        role="tab"
        tabIndex={null}
      >
        barTab
      </li>
    </ul>
    <div
-     aria-labelledby="react-tabs-4"
+     aria-labelledby="react-tabs-8"
      className="react-tabs__tab-panel react-tabs__tab-panel--selected"
-     id="react-tabs-5"
+     id="react-tabs-9"
      role="tabpanel"
    >
      fooPanel
    </div>
    <div
-     aria-labelledby="react-tabs-6"
+     aria-labelledby="react-tabs-10"
      className="react-tabs__tab-panel"
-     id="react-tabs-7"
+     id="react-tabs-11"
      role="tabpanel"
    />
  </div>
`;

exports[`Tabs toMatchSnapshot 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  max-width: 1140px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding: 0;
  margin-bottom: 15px;
  overflow: auto hidden;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 185px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  font-size: 14px;
  text-transform: uppercase;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  padding: 5px;
  background-color: rgb(178,178,178);
  border: 1px solid rgb(0,0,0);
  border-left: none;
  border-bottom: none;
  cursor: pointer;
}

.c2.selected {
  background-color: rgb(242,242,242);
  padding: 7px 5px;
  border-top: 4px solid rgb(0,0,0);
  border-bottom: 1px solid rgb(242,242,242);
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
  cursor: default;
}

.c2.selected:first-child {
  border-left: 1px solid;
}

.c2.disabled {
  cursor: not-allowed;
}

@media (min-width:767px) {
  .c1 {
    margin-bottom: 30px;
  }
}

@media (min-width:767px) {
  .c2 {
    -webkit-flex-flow: row;
    -ms-flex-flow: row;
    flex-flow: row;
    font-size: 18px;
    padding: 10px 15px;
  }
}

@media (min-width:767px) {
  .c2.selected {
    padding: 15px;
  }
}

<div
  className="c0"
  data-tabs={true}
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <ul
    className="c1"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      className="c2 selected"
      id="react-tabs-0"
      role="tab"
      tabIndex="0"
    >
      fooTab
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      className="c2"
      id="react-tabs-2"
      role="tab"
      tabIndex={null}
    >
      barTab
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-1"
    role="tabpanel"
  >
    fooPanel
  </div>
  <div
    aria-labelledby="react-tabs-2"
    className="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
  />
</div>
`;
